Articles/FileSystemProviders/extending.md

```diff
@@ -0,0 +1,213 @@
+---
+v8-equivalent: "https://our.umbraco.com/documentation/Extending/FileSystemProviders/"
+versionFrom: 9.0.0
+verified-against: beta-1
+---
+
+# Custom file systems (IFileSystem)
+
+## Media Filesystem
+By default, Umbraco uses an instance of `PhysicalFileSystem` to handle the storage location of the media archive (wwwroot/media).
+
+This can be configured during composition:
+
+```csharp
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Umbraco.Cms.Core.Composing;
+using Umbraco.Cms.Core.DependencyInjection;
+using Umbraco.Cms.Core.Hosting;
+using Umbraco.Cms.Core.IO;
+using Umbraco.Cms.Infrastructure.DependencyInjection;
+
+namespace UmbracoExamples.Composition
+{
+    public class SetMediaFileSystemComposer : IUserComposer
+    {
+        public void Compose(IUmbracoBuilder builder)
+        {
+            builder.SetMediaFileSystem((factory) =>
+            {
+                IHostingEnvironment hostingEnvironment = factory.GetRequiredService<IHostingEnvironment>();
+                var folderLocation = "~/CustomMediaFolder";
+                var rootPath = hostingEnvironment.MapPathWebRoot(folderLocation);
+                var rootUrl = hostingEnvironment.ToAbsolute(folderLocation);
+
+                return new PhysicalFileSystem(
+                    factory.GetRequiredService<IIOHelper>(),
+                    hostingEnvironment,
+                    factory.GetRequiredService<ILogger<PhysicalFileSystem>>(),
+                    rootPath,
+                    rootUrl);
+            });
+        }
+    }
+```
+
+When creating a PhyscialFileSystem it takes some dependencies like IIOHelper, but the last two parameters are what we're interested in. The rootPath is where your media will be stored on the disk, since netcore by default store files in the wwwroot, we must put our desired folder somewhere within wwwroot, to ensure that we use `hostingEnvironment.MapPathWebRoot(~/CustomMediaFolder)`, the ~ will be mapped to your wwwroot folder, so the final rootPath will be your/project/path/wwwroot/CustomMediaFolder, the ~ is therefore important!
```

Comment:
Split things up to more "edible" sentences, and added some extra formatting as well.
Hope I haven't ruined any of the meaning in this section - let me know!

```suggestion
When creating a PhyscialFileSystem it takes some dependencies like IIOHelper, but the last two parameters are what we're interested in. The `rootPath` is where your media will be stored on the disk. Since netcore by default store files in the `wwwroot`, we must put our desired folder somewhere within `wwwroot`, to ensure that we use `hostingEnvironment.MapPathWebRoot(~/CustomMediaFolder)`. The `~` will be mapped to your `wwwroot` folder, so the final `rootPath` will be `your/project/path/wwwroot/CustomMediaFolder`. The `~` is therefore important.
```